### ISSUE TITLE
[Bug]: CI/CD workflows are vulnerable to PWN request attack

### ISSUE THREAD

**Issue by sakul1234** [matrixorigin/matrixone#12429]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Your current CI/CD configuration allows all external pull request creators to run actions on untrusted code. You can read about this type of attack here - https://securitylab.github.com/research/github-actions-preventing-pwn-requests. 

Specifically this applies to workflow inside this file - https://github.com/matrixorigin/matrixone/blob/main/.github/workflows/ci.yml.

This means that all defined secrets and credentials can be exfiltrated by any external contributor.

### Expected Behavior

All external PRs should be approved manually before running any action on the untrusted code.

To remediate the issue, you can:

- Avoid using pull_request_target if the workflow doesn’t need write repository permissions and doesn’t use any repository secrets. They can simply use the pull_request trigger instead.

- Assign repository privileges only where needed explicitly through pull_request and workflow_run as in our previous example.

- Add a condition to the pull_request_target to run only if a certain label is assigned the PR, like safe to test that indicates the PR has been vetted by someone with write privileges to the target repository. Note that this kind of label based verification is still prone to a race condition in which the attacker may push new changes after the workflow was approved (labeled), but has not started yet. As such this approach should only be used as a temporary solution, until a proper fix from the options above is applied. Since external users do not have the permission to assign labels, this effectively requires repository owners to manually review changes first and is also prone to human error.

 ```
   # Only as a temporary fix.
    on:
      pull_request_target:
        types: [labeled]

    jobs:
      build:
        name: Build and test
        runs-on: ubuntu-latest
        if: contains(github.event.pull_request.labels.*.name, 'safe to test')
```

### Steps to Reproduce

```Markdown
Just creating a PR is enough. Sample PoC was done inside this PR - https://github.com/matrixorigin/matrixone/pull/12426
```

### Additional information

**In the end, please rotate existing keys and credentials because of the possible compromise.**

**Comment by guguducken** [comment]
modified in the pr #12438.
Details as follows:
1. Separate the jobs that need to use secrets into an action `Matrixone Utils CI`.
2. If the creator of pull request is the member of `matrixorigin`, pull_request_target action will run automatically.
3. others, the pull_request_target action will not run unless there is a specific label `safe-to-test` on this pr

**Comment by guguducken** [comment]
fixed in pr #12438
